src/libgap-api.c

```diff
@@ -0,0 +1,337 @@
+// LibGAP API - API for using GAP as shared library.
+
+#include "libgap-api.h"
+
+#include "bool.h"
+#include "calls.h"
+#include "gapstate.h"
+#include "gvars.h"
+#include "intobj.h"
+#include "opers.h"
+#include "plist.h"
+#include "stringobj.h"
+
+Obj FuncREAD_ALL_COMMANDS(
+    Obj self, Obj instream, Obj echo, Obj capture, Obj resultCallback);
+
+// Setup and initialisation
+void GAP_Initialize(int          argc,
+                    char **      argv,
+                    char **      env,
+                    CallbackFunc markBagsCallback,
+                    CallbackFunc errorCallback)
+{
+    InitializeGap(&argc, argv, env);
+    SetExtraMarkFuncBags(markBagsCallback);
+    STATE(JumpToCatchCallback) = errorCallback;
+}
+
+// TODO: This function is far from completely finalizing a GAP session
+//       it potentially has to close all open file descriptors, kill
+//       subprocesses, etc
+void GAP_Finalize(void)
+{
+    FinishBags();
+}
+
+void GAP_ClearError(void)
+{
+    ClearError();
+}
+
+void GAP_Collect(void)
+{
+    CollectBags(0, 1);
+}
+
+const char * GAP_GetTNam(Obj val)
+{
+    return TNAM_OBJ(val);
+}
+
+uint64_t GAP_GetTNum(Obj val)
+{
+    return TNUM_OBJ(val);
+}
+
+
+//
+// Booleans
+//
+bool GAP_IsBoolean(Obj val)
+{
+    return TNUM_OBJ(val) == T_BOOL;
+}
+
+// What about fail :/
+Obj GAP_MakeBoolean(bool val)
```

Comment:
So are we now using `bool`? A previous PR of mine suggesting that was shot down.

I am actually confused as to why this code compiles -- I thought to access `bool` in C99 code, one has to `#include <stdbool.h>` first? Are we sure this code is not relying on a GCC extension then (which might make `bool` unconditionally available in C code)?